ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -616,6 +625,83 @@ private void readFromFile(File fileToRead, Journal journal) throws StoreExceptio
   }
 
   /**
+   * Reads and constructs the index segment from the DataInputStream and CrcInputStream
+   * @param stream the {@link DataInputStream} that needs to be read
+   * @param crcStream the {@link CrcInputStream} from which crc needs to be read
+   * @param journal the {@link Journal} to use to populate new entries
+   * @throws IOException
+   * @throws StoreException
+   */
+  private void readFromStream(DataInputStream stream, CrcInputStream crcStream, Journal journal)
+      throws IOException, StoreException {
```

Comment:
Can you provide me some examples of where we have separate paths even though most part (in this case around 90%) is same. I feel this can be merged to one block as it is just two additional fields and the rest all remains the same.